crates/goose/src/providers/formats/databricks.rs

```
@@ -246,15 +246,12 @@ pub fn format_tools(tools: &[Tool]) -> anyhow::Result<Vec<Value>> {
             return Err(anyhow!("Duplicate tool name: {}", tool.name));
         }
 
-        let mut description = tool.description.clone();
-        description.truncate(1024);
-
-        // OpenAI's tool description max str len is 1024
         result.push(json!({
             "type": "function",
             "function": {
                 "name": tool.name,
-                "description": description,
+                // do not silently truncate description
+                "description": tool.description,
```

I think could treat databricks seperately as we don't know if all models support that beyond openai via model serving on databricks

---

The point of this PR is also to say that it is pretty much never a good idea to silently truncate as it yields very subtlely wrong behavior that is really annoying to debug. So if other models via databricks do not support more than 1024, then I think it is more desirable for users to see the error.

Or for each provider, detect if the problem came from a too long context, and provide a nicer/actionable error message to the user.